src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -841,6 +826,15 @@ private void handleFailure(AzureVMAgentTemplate template, String vmName, Excepti
 
         LOGGER.log(Level.INFO,
                 "AzureVMCloud: provision: asynchronous provision finished, returning {0} planned node(s)", plannedNodes.size());
+
+        final Map<String, String> properties = new HashMap<>();
+        properties.put("NumberOfAgents", String.valueOf(plannedNodes.size()));
+        properties.put(AppInsightsConstants.AZURE_SUBSCRIPTION_ID,
+                template.getAzureCloud().getServicePrincipal().getSubscriptionId());
+        properties.put(AppInsightsConstants.AZURE_LOCATION, template.getLocation());
+        properties.put("AgentOS", template.getOsType());
+        AzureVMAgentPlugin.sendEvent(Constants.AI_VM_AGENT, "Provision", properties);
```

Comment:
It's  a long method with many `try{..}catch{...}` blocks.  Will try my best to find some place to send an event but cannot make sure it covers all branches.